Assets/CodeBase/Entities/player/Player.cs

```diff
@@ -78,27 +94,70 @@ void Start()
         inputProfile.addListener(InputEvent.Key, PlayerInputProfile.moveDown, moveDown);
         inputProfile.addListener(InputEvent.Up, PlayerInputProfile.moveUp, stopClimbing);
         inputProfile.addListener(InputEvent.Up, PlayerInputProfile.moveDown, stopClimbing);
-        inputProfile.addListener(InputEvent.Key, PlayerInputProfile.jump, jump);
+        inputProfile.addListener(InputEvent.Down, PlayerInputProfile.jump, jump);
 
         //Listeners for abilities.
         inputProfile.addListener(InputEvent.Up, PlayerInputProfile.toggleIce, toggleIce);
         inputProfile.addListener(InputEvent.Up, PlayerInputProfile.toggleFire, toggleFire);
         inputProfile.addListener(InputEvent.Up, PlayerInputProfile.toggleWind, toggleWind);
         inputProfile.addListener(InputEvent.Up, PlayerInputProfile.toggleEarth, toggleEarth);
 
+        //shift key
+        inputProfile.addListener(InputEvent.Down, PlayerInputProfile.shift, shift);
+
         currentAbility = ActiveAbility.NORMAL;
         this.enabled = false;
         Controller.instance.stateMachine.AddStateListener(onStateChange);
 
     }
 
     // Update is called once per frame
-    void Update()
+    void FixedUpdate()
```

Comment:
Keep in mind that FixedUpdate() is better for updating physics related components while Update() is more optimal for detecting user input.